tools/bin/hawqsync-falcon

```diff
@@ -0,0 +1,1331 @@
+        sys.exit(1)
+
+    # ### Leave safemode on source HDFS
+    print "Disable HDFS safemode on source cluster" if options.verbose else None;
+    retVal, stderr = disableHdfsSafemode(isTesting=options.testMode)
+
+    if retVal != 0:
+        print "Failed to leave HDFS safemode on source cluster"
+        print "Error message was " + stderr
+        print "Exiting"
+        sys.exit(1)
+
+    # ### Force HDFS checkpoint and enable safemode on target
+    print "Enabling HDFS safemode on target cluster" if options.verbose else None
+    retVal, stderr = enableHdfsSafemode(namenodeHost=options.targetNamenode, isTesting=options.testMode)
+
+    if retVal != 0:
+        print "Failed to enable HDFS safemode on target cluster"
+        print "Error message was " + stderr
+        print "Exiting"
+        sys.exit(1)
+
+    print "Forcing HDFS checkpoint on target cluster" if options.verbose else None
+    retVal, stderr = forceHdfsCheckpoint(namenodeHost=options.targetNamenode, isTesting=options.testMode)
+
+    if retVal != 0:
+        print "Failed to force HDFS checkpoint on target cluster"
+        print "Error message was " + stderr
+        print "Exiting"
+        sys.exit(1)
+
+    # ### HDFS Fingerprint comparison
+
+    print "Validating HDFS fingerprints match between source and target clusters" if options.verbose else None
+
+    retVal, md5OrStderr = getHdfsFingerprint(hdfsUri=jobParameters['sourceHdfsUri'],
+                                             hdfsDir=jobParameters['pathToSync'],
+                                             isTesting=options.testMode)
+
+    if retVal != 0:
+        print "Failed to generate HDFS fingerprint on source cluster"
+        print "Error message was " + md5OrStderr
+        print "Exiting"
+        sys.exit(1)
+
+    retVal, md5OrStderr2 = getHdfsFingerprint(hdfsUri=jobParameters['targetHdfsUri'],
+                                              hdfsDir=jobParameters['pathToSync'],
+                                              isTesting=options.testMode)
+
+    if retVal != 0:
+        print "Failed to generate HDFS fingerprint on target cluster"
+        print "Error message was " + md5OrStderr2
+        print "Exiting"
+        sys.exit(1)
+
+    isValidSync = hdfsFingerprintsMatch(md5OrStderr, md5OrStderr2)
+
+    if not isValidSync:
+        print "Source and target cluster HDFS fingerprints do not match."
+        print "Source checksum : " + md5OrStderr
+        print "Target checksum : " + md5OrStderr2
+        print "This is bad, please check Falcon sync logs : " + falconOutput
+        sys.exit(1)
+    else:
+        print "Source and target HDFS fingerprints match."
+
+    # Target
+    # TODO - either use Ambari REST call or use SSH
+    print "Starting target HAWQ" if options.verbose else None;
+    retVal, stderr = startHawq(masterHost=options.targetHawqMaster,
+                               isTesting=options.testMode)
+    print retVal, stderr if options.verbose else None;
+
```

Comment:
@kdunn926 , may I understand till here all the files are copied to new HAWQ master data directory, but HAWQ catalog information has not been changed. We can leverage hawq register to register the files into HAWQ?